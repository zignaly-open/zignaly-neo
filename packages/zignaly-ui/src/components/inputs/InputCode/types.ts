export type InputCodeProps = {
  fields: number;
  onComplete: any;
  loading: boolean;
  clearOnError?: boolean;
  error?: string | null;
  autoFocus?: boolean;
};
