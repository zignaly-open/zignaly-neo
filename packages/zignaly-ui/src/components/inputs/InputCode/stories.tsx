// Dependencies
import * as React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

// Component
import InputCode from "./";

export default {
  title: "Inputs/InputCode",
  component: InputCode,
  argTypes: {
    fields: {
      type: "number",
    },
    onComplete: {
      type: "function",
    },
    loading: {
      type: "boolean",
      defaultValue: false,
    },
  },
} as ComponentMeta<typeof InputCode>;

const Template: ComponentStory<typeof InputCode> = (args) => <InputCode {...args} />;

export const Basic = Template.bind({});
Basic.args = {
  fields: 6,
  loading: false,
  onComplete: () => {},
};
