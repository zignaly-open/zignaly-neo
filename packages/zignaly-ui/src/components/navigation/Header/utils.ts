// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
