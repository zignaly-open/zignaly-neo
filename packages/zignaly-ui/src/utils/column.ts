import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
