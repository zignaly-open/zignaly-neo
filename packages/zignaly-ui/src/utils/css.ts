import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
