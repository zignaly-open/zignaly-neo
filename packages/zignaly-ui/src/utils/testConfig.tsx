import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
