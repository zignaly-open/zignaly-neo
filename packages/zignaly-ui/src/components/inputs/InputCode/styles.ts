import styled from "styled-components";
import { styledIf } from "../../../utils/styled";

export const Layout = styled.div<{ error?: string }>`
  display: flex;
  width: 100%;
  flex-direction: column;
  .input-box {
    background: #0f1124;

    input {
      background: #0f1124;
      background: linear-gradient(90deg, #0f1124 0%, #0f1124 35%, #0f1124 100%);
      border-color: #35334a !important;
      color: ${(props) => `${props.theme.neutral100}`};
      font-family: "Avenir Next", "Red Hat Text", sans-serif;
    }
  }

  ${(props) => `
    ${styledIf(
      props.error,
      `
       .input-box {
          margin-bottom: 10px;
          input {
            border-color: ${props.theme.redGraphOrError} !important;
          }
       }
    `,
    )}
  `}
`;
