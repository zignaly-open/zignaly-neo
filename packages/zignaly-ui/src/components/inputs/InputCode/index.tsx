import React, { useEffect, useRef } from "react";
import ReactCodeInput from "@zignaly-open/react-verification-code-input";
import { InputCodeProps } from "./types";
import { Layout } from "./styles";
import ErrorMessage from "components/display/ErrorMessage";
import { Box } from "@mui/material";

function InputCode({
  fields,
  loading,
  onComplete,
  autoFocus = false,
  clearOnError = false,
  error = null,
  prefixId,
}: InputCodeProps) {
  const inputRef = useRef<ReactCodeInput>(null);

  useEffect(() => {
    if (inputRef.current) {
      const input = inputRef.current as any;
      if (input.iRefs.length && autoFocus) {
        setTimeout(() => {
          input.iRefs[0].current?.focus();
        }, 300); // TODO: figure out why
      }
    }
  }, [inputRef]);

  useEffect(() => {
    if (error) {
      // means we just received an error
      // erroneous code => type again
      // private methods, you say? sorry, ya vas ne ponimat, lol
      clearOnError && inputRef.current?.__clearvalues__();
    }
  }, [error, clearOnError]);

  return (
    <Layout error={error || undefined}>
      <Box id={prefixId && `${prefixId}__code-input`}>
        <ReactCodeInput
          ref={inputRef}
          className={"input-box"}
          fields={fields}
          loading={loading}
          disabled={loading}
          autoFocus={autoFocus}
          onComplete={onComplete}
        />
      </Box>
      {error && <ErrorMessage text={error} id={prefixId && `${prefixId}__error-message`} />}
    </Layout>
  );
}

export default InputCode;
