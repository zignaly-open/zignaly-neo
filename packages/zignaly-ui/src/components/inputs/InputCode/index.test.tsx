import React from "react";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, waitFor } from "@testing-library/react";
import InputCode from ".";

describe("components/inputs/InputCode", () => {
  it("should render without crashing", () => {
    const { container } = renderWithProvidersUi(
      <InputCode prefixId={"test"} fields={6} loading={false} onComplete={jest.fn()} />,
    );

    const inputs = container.querySelector("#test__code-input") as Element;
    expect(inputs).toBeVisible();
    const inputsStyles = getComputedStyle(inputs);
    expect(inputs).toMatchSnapshot();
    expect(inputsStyles).toMatchSnapshot();
  });

  it("should dispatch onComplete when all input fields are filled", () => {
    const handleOnComplete = jest.fn();
    const { container } = renderWithProvidersUi(
      <InputCode prefixId={"test"} fields={6} loading={false} onComplete={handleOnComplete} />,
    );
    const inputFields = container.getElementsByTagName("input");
    Array.from(inputFields).forEach((inputFields) => {
      fireEvent.change(inputFields, { target: { value: "1" } });
    });
    waitFor(() => {
      expect(handleOnComplete).toHaveBeenCalled();
    });
  });

  it("should display error message when `error` prop is passed", () => {
    const { container } = renderWithProvidersUi(
      <InputCode
        prefixId={"test"}
        fields={6}
        loading={false}
        onComplete={jest.fn()}
        error="Test error"
      />,
    );

    const errorMessage = container.querySelector("#test__error-message") as Element;
    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toHaveTextContent("Test error");
  });
});
