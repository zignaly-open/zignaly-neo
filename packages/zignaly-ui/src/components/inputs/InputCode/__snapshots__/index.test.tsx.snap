// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/inputs/InputCode should render without crashing 1`] = `
<div
  class="MuiBox-root css-0"
  id="test__code-input"
>
  <div
    class="styles_react-code-input-container__tpiKG input-box"
    style="width: 348px;"
  >
    <div
      class="styles_react-code-input__CRulA"
    >
      <input
        autocomplete="one-time-code"
        data-id="0"
        pattern="[0-9]*"
        placeholder=" "
        style="width: 58px; height: 54px;"
        type="tel"
        value=""
      />
      <input
        autocomplete="one-time-code"
        data-id="1"
        pattern="[0-9]*"
        placeholder=" "
        style="width: 58px; height: 54px;"
        type="tel"
        value=""
      />
      <input
        autocomplete="one-time-code"
        data-id="2"
        pattern="[0-9]*"
        placeholder=" "
        style="width: 58px; height: 54px;"
        type="tel"
        value=""
      />
      <input
        autocomplete="one-time-code"
        data-id="3"
        pattern="[0-9]*"
        placeholder=" "
        style="width: 58px; height: 54px;"
        type="tel"
        value=""
      />
      <input
        autocomplete="one-time-code"
        data-id="4"
        pattern="[0-9]*"
        placeholder=" "
        style="width: 58px; height: 54px;"
        type="tel"
        value=""
      />
      <input
        autocomplete="one-time-code"
        data-id="5"
        pattern="[0-9]*"
        placeholder=" "
        style="width: 58px; height: 54px;"
        type="tel"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`components/inputs/InputCode should render without crashing 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "visibility",
  "_importants": {
    "display": "",
    "visibility": undefined,
  },
  "_length": 2,
  "_onChange": [Function],
  "_values": {
    "display": "block",
    "visibility": "visible",
  },
}
`;
